### PR TITLE
Add auth0 guide for securing APIs.

### PR DESCRIPTION
**Pull request by tjholm**

### CODE REVIEW

**Comment by HomelessDinosaur** [comment]
Should Auth0 be capitalised throughout the guide?

File: docs/guides/secure-api-auth0.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
+---
+title: Securing your API with auth0
+description: Configure an Auth0 application and secure your API with it
+---
+
+## What we'll be doing
+
+In this guide we'll use the Nitric Framework with an [auth0](https://auth0.com/) application to build serverless functions and a secure API.
+
+This example adds secure JWT authentication to an API allowing you to integrate it either with standalone api login, or to integrate it with a frontend application using the clients identity token.
```
Comment (HomelessDinosaur) [inline docs/guides/secure-api-auth0.md:10]: client's

File: docs/guides/secure-api-auth0.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+First we'll need the `audience` for our API, this can be found on the settings tab of your new API.
+
+<img
+  alt="get audience value"
+  src="../../assets/img/guides/auth0/auth0-get-audience.png"
+  height="700"
+  />
+
+Second we'll get the `issuer` for our API, this will be our auth0 environment endpoint and easy way to see this is to navigate the the `Test` tab our our API.
```
Comment (HomelessDinosaur) [inline docs/guides/secure-api-auth0.md]: ...endpoint. An easy way to see this is to navigate to the `Test` tab of our API.

File: docs/guides/secure-api-auth0.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+helloApi.get('/hello/:name', async (ctx) => {
+  const { name } = ctx.req.params;
+
+  ctx.res.body = `Hello ${name}`;
+
+  return ctx;
+});
+```
+
```
Comment (HomelessDinosaur) [inline docs/guides/secure-api-auth0.md:108]: ## Testing the Security Definition

File: docs/guides/secure-api-auth0.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+  />
+
+Second we'll get the `issuer` for our API, this will be our auth0 environment endpoint and easy way to see this is to navigate the the `Test` tab our our API.
+
+<img
+  alt="get audience value"
+  src="../../assets/img/guides/auth0/auth0-get-issuer.png"
+  height="700"
+  />
+
```
Comment (HomelessDinosaur) [inline docs/guides/secure-api-auth0.md:61]: ## Configuring the Nitric API

**Review by HomelessDinosaur — CHANGES_REQUESTED** [review]

File: docs/guides/secure-api-auth0.md (hunk 2 above)
Comment (tjholm) [reply docs/guides/secure-api-auth0.md]: fixed.

File: docs/guides/secure-api-auth0.md (hunk 1 above)
Comment (tjholm) [reply docs/guides/secure-api-auth0.md]: fixed.

File: docs/guides/secure-api-auth0.md (hunk 4 above)
Comment (tjholm) [reply docs/guides/secure-api-auth0.md:63]: Header added

File: docs/guides/secure-api-auth0.md (hunk 3 above)
Comment (tjholm) [reply docs/guides/secure-api-auth0.md:110]: Header added.

**Review by HomelessDinosaur — APPROVED** [review]

**Review by jyecusch — APPROVED** [review]

**Pull request merged**